CRM/Core/Resources.php

```diff
@@ -809,4 +810,25 @@ public static function getEntityRefFilters() {
     return $filters;
   }
 
+  /**
+   * @param string $url
+   * @return string
+   */
+  public function addCacheCode($url) {
+    parse_str(parse_url($url, PHP_URL_QUERY), $queryParts);
```

Comment:
Or let say this: I'd do `strpos()` (or maybe keep `getUrl()` unchanged). But empirically the current should be acceptable at this scale.